storagemarket/deal_execution.go

```diff
@@ -685,66 +687,88 @@ func (p *Provider) registerShardSync(ctx context.Context, pc cid.Cid) error {
 
 // fireSealingUpdateEvents periodically checks the sealing status of the deal
 // and fires events for each change
-func (p *Provider) fireSealingUpdateEvents(dh *dealHandler, dealUuid uuid.UUID, sectorNum abi.SectorNumber) {
-	var lastSealingState lapi.SectorState
-	checkStatus := func(force bool) lapi.SectorState {
+func (p *Provider) fireSealingUpdateEvents(dh *dealHandler, dealUuid uuid.UUID, sectorNum abi.SectorNumber) *dealMakingError {
+	var sectorInfo lapi.SectorInfo
+	var deal *types.ProviderDealState
+	checkStatus := func(force bool) lapi.SectorInfo {
 		// To avoid overloading the sealing service, only get the sector status
 		// if there's at least one subscriber to the event that will be published
 		if !force && !dh.hasActiveSubscribers() {
-			return ""
+			return lapi.SectorInfo{}
 		}
 
 		// Get the sector status
 		si, err := p.sps.SectorsStatus(p.ctx, sectorNum, false)
-		if err == nil && si.State != lastSealingState {
-			lastSealingState = si.State
+		if err == nil && si.State != sectorInfo.State {
+			sectorInfo.State = si.State
```

Comment:
This is a little tricky to understand - I'd suggest keeping the lastSealingState variable and leaving these lines as they were